### DISCUSSION TITLE
Question about property syncing without ownership

### DISCUSSION THREAD

**Discussion by Netherite1701** [foxssake/netfox#219] [Q&A] [answered]
Hi, I have a Netfox project with 3d player movement. The movement script changes the player's rotation but ownership blocks it from turning. How would I go about solving this?

(Sorry if this is formatted improperly. I'm new to github)

Below is the project.
[netfox.zip](https://github.com/foxssake/netfox/files/14893565/netfox.zip)

**Comment by elementbound** [comment]
Hey @Netherite1701, thanks for attaching a project, much appreciated! Which scene should I look at?

**Reply by Netherite1701** [reply]
Oh, @elementbound ! Thanks for the reply!

I'm currently working on the 'multiplayer-netfox' scene and my player is on 'scenes/player.tscn'.

**Reply by elementbound** [reply, marked as answer]
Ok, I looked into it and I get what's your issue - the head rotation is synchronized by a `MultiplayerSynchronizer` node, but it gets reset on clients, because the player avatar's owner is the server.

I would recommend moving the mouselook logic to the rollback loop, same as with player movement.

To do so, gather the mouse movement into a property on `PlayerInput`. Then, in the player's _rollback_tick, apply that. Make sure to configure both the mouse input and the player- and neck rotation as input and state properties in the `RollbackSynchronizer`. 

This would also mean that the mouselook would be tied to the network tickrate even for the local player. To counteract this, you should be able to keep the existing mouselook logic in the player class as well, only running it if the avatar belongs to the local player. What should happen is that the `PlayerInput` class gathers the mouselook rotation, which then gets applied in the rollback tick, and inbetween network ticks, the same rotations are applied as they arrive on the player avatar, so rotation feels instant.
